### PR TITLE
Convert to new logging convention: spider.log -> spider.logger

### PR DESCRIPTION
**Pull request by chishaku**

### CODE REVIEW

**Comment by Germey** [comment]
What's the difference or advantange?

**Comment by LuckyPigeon** [comment]
@Germey 
It just an update of different version, before scrapy 1.0, they use `spider.log` for logging message, and it has been deprecated after 1.0. The new logging from spider can be called by `spider.logger`.

**Review by LuckyPigeon — DISMISSED** [review]

**Comment by Germey** [comment]
@LuckyPigeon But seems the `log` function is still kept in Spider: https://github.com/scrapy/scrapy/blob/c5ab58056c29c2c35b183572b0780acfbb15dfe8/scrapy/spiders/__init__.py#L37

As the document (https://docs.scrapy.org/en/latest/topics/spiders.html#scrapy.spiders.Spider.log) said: 
`log` function is the wrapper that sends a log message through the Spider’s logger, kept for backward compatibility.

@rmax What's your opinion?

**Comment by rmax** [comment]
When was this change introduced?

That will change the minimum scrapy version required which is not a bad thing. 

Probably you may want to think on the next major releases supporting python >=3.7 and corresponding minimum scrapy version deprecating what needs to be deprecated.

**Comment by Germey** [comment]
@rmax For this PR, I don't see the deprecation of `log` function of Spider class according to the document, so I think the change of this PR is not required.

> Probably you may want to think on the next major releases supporting python >=3.7 and corresponding minimum scrapy version deprecating what needs to be deprecated.
For this, it's a good idea, I can also help think about it. For current Scrapy's version, this pacakge is compatiable.

File: tests/test_scrapy_redis.py [hunk 1]
```diff
@@ -193,7 +193,7 @@ def test_scheduler(self):
 
     def test_scheduler_persistent(self):
         # TODO: Improve this test to avoid the need to check for log messages.
-        self.spider.log = mock.Mock(spec=self.spider.log)
+        self.spider.log = mock.Mock(spec=self.spider.logger)
```
Comment (Germey) [inline tests/test_scrapy_redis.py:196]: I think is better to keep log function here, something wired to use `log = ... logger`

Comment (LuckyPigeon) [reply tests/test_scrapy_redis.py:196]: @Germey 
I think you're right, we should keep log function here for the time being.

**Comment by LuckyPigeon** [comment]
@chishaku 
Although in the logging section of scrapy doc recommends to use `spider.logger` for logging message, but they still kept a backward compatibility for `spider.log`.
For the time being, we won't merge this request, but thanks for your work! Looking forward to see your contributions.

File: tests/test_scrapy_redis.py (hunk 1 above)
Comment (LuckyPigeon) [reply tests/test_scrapy_redis.py:196]: I've tried to use `log` and `logger`, they both work. 
But there's a little different, `log` utility need to import `logging` to show other types of message, or it only shows `DEBUG` message. One the other hand, `logger` utility can show all kind of message without import a third party library.

Comment (LuckyPigeon) [reply tests/test_scrapy_redis.py:196]: But the usage of line 196 is wrong, should be `spider.logger.<log level>`.

**Comment by LuckyPigeon** [comment]
@Germey @rmax 
> Probably you may want to think on the next major releases supporting python >=3.7 and corresponding minimum scrapy version deprecating what needs to be deprecated.
Let's discuss this in another issue.
Close this PR due to not merging.

**Pull request closed without merging**